crates/oxc_linter/src/rules/eslint/no_lone_blocks.rs

```diff
@@ -0,0 +1,374 @@
+use oxc_ast::ast::{Declaration, VariableDeclarationKind};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_lone_blocks_diagnostic(span: Span, is_nested_block: bool) -> OxcDiagnostic {
+    let message =
+        if is_nested_block { "Nested block is redundant." } else { "Block is unnecessary." };
+    // See <https://oxc.rs/docs/contribute/linter/adding-rules.html#diagnostics> for details
+    OxcDiagnostic::warn(message).with_label(span)
+}
```

Comment:
for clarity, i would suggest splitting it into two diagnostic messages.

e.g. `fn no_lone_blocks_diagnostic` and `fn no_nested_lone_blocks_diagnostic`

Reply:
updated! https://github.com/oxc-project/oxc/pull/8145/commits/d272fe594a55eb9f14561c6ce105a9d2cb85a1e0